module.go

```diff
@@ -1,137 +1,97 @@
+	_, span := trace.StartSpan(ctx, "service::vision::register3DSegmenterFromDetector")
+	defer span.End()
+	if conf == nil {
+		return nil, errors.New("config for 3D segmenter made from a detector cannot be nil")
+	}
+	detectorService, err := vision.FromDependencies(deps, conf.DetectorName)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "could not find necessary dependency, detector %q", conf.DetectorName)
 	}
-
-	return NewDetectionsToSegments(ctx, deps, rawConf.ResourceName(), conf, logger)
-
-}
-
-func NewDetectionsToSegments(ctx context.Context, deps resource.Dependencies, name resource.Name, conf *Config, logger logging.Logger) (vision.Service, error) {
-
-	cancelCtx, cancelFunc := context.WithCancel(context.Background())
-
-	s := &detectionsToSegmentsDetectionsToSegments{
-		name:       name,
-		logger:     logger,
-		cfg:        conf,
-		cancelCtx:  cancelCtx,
-		cancelFunc: cancelFunc,
+	confThresh := 0.5 // default value
+	if conf.ConfidenceThresh > 0.0 {
+		confThresh = conf.ConfidenceThresh
 	}
-	return s, nil
-}
-
-func (s *detectionsToSegmentsDetectionsToSegments) Name() resource.Name {
-	return s.name
-}
-
-func (s *detectionsToSegmentsDetectionsToSegments) NewClientFromConn(ctx context.Context, conn rpc.ClientConn, remoteName string, name resource.Name, logger logging.Logger) (vision.Service, error) {
-	panic("not implemented")
-}
-
-func (s *detectionsToSegmentsDetectionsToSegments) DetectionsFromCamera(ctx context.Context, cameraName string, extra map[string]interface{}) ([]objdet.Detection, error) {
-	panic("not implemented")
-}
-
-func (s *detectionsToSegmentsDetectionsToSegments) Detections(ctx context.Context, img image.Image, extra map[string]interface{}) ([]objdet.Detection, error) {
-	panic("not implemented")
-}
-
-func (s *detectionsToSegmentsDetectionsToSegments) ClassificationsFromCamera(ctx context.Context, cameraName string, n int, extra map[string]interface{}) (classification.Classifications, error) {
-	panic("not implemented")
+	detector := func(ctx context.Context, img image.Image) ([]objectdetection.Detection, error) {
+		return detectorService.Detections(ctx, img, nil)
+	}
+	segmenter, err := DetectionSegmenter(objectdetection.Detector(detector), conf.MeanK, conf.Sigma, confThresh)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create 3D segmenter from detector")
+	}
+	if conf.DefaultCamera != "" {
+		_, err = camera.FromDependencies(deps, conf.DefaultCamera)
+		if err != nil {
+			return nil, errors.Errorf("could not find camera %q", conf.DefaultCamera)
+		}
+	}
+	return vision.NewService(name, deps, logger, nil, nil, detector, segmenter, conf.DefaultCamera)
 }
 
-func (s *detectionsToSegmentsDetectionsToSegments) Classifications(ctx context.Context, img image.Image, n int, extra map[string]interface{}) (classification.Classifications, error) {
-	panic("not implemented")
-}
+func (conf DetectionSegmenterConfig) Validate(path string) ([]string, []string, error) {
```

Comment:
I'd put all methods defined on a struct in the same file where the struct itself is defined. Otherwise, it can become confusing to find what methods exist and where they are defined, and maybe you move code into a subdirectory in the future and it breaks because you moved the struct but not the method or vice versa.